src/models/rec.ts

```diff
@@ -1,15 +1,64 @@
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+import customParseFormat from "dayjs/plugin/customParseFormat";
 import { RecoveryRecord, RecRecord } from "../types";
 import { targetDisease } from "./pdt";
 
+dayjs.locale("en-sg");
+dayjs.extend(utc);
+dayjs.extend(customParseFormat);
+
 /**
  * Helper function to generate a Recovery record and ensures that testResultCode has correct EU record.
  * @param recoveryRecord
  * @returns
  */
 export const genRecoveryRecord = (recoveryRecords: RecoveryRecord[], expiryDateTime: string, issuerName: string, reference: string): RecRecord[] => {
-  
+  const validatedRecoveryRecords = (recoveryRecords).map((recovery) => {
+    const validateFirstPositiveTestDate = recovery.firstPositiveTestDate.split('T')[0]; // take only date value if ISO-8601 format
+    if (
+      !dayjs(
+        validateFirstPositiveTestDate,
+        ["YYYY-MM-DD"],
+        true
+      ).isValid()
+    ) {
+      throw new Error(
+        `Invalid firstPositiveTestDate (${recovery.firstPositiveTestDate}). Should be YYYY-MM-DD format`
+      );
+    }
+    const validateTestValidFrom = recovery.testValidFrom.split('T')[0]; // take only date value if ISO-8601 format
+    if (
+      !dayjs(
+        validateTestValidFrom,
+        ["YYYY-MM-DD"],
+        true
+      ).isValid()
+    ) {
+      throw new Error(
+        `Invalid testValidFrom (${recovery.testValidFrom}). Should be YYYY-MM-DD format`
+      );
+    }
+    const validateTestValidUntil = recovery.testValidUntil.split('T')[0]; // take only date value if ISO-8601 format
+    if (
+      !dayjs(
+        validateTestValidUntil,
+        ["YYYY-MM-DD"],
+        true
+      ).isValid()
+    ) {
+      throw new Error(
+        `Invalid testValidUntil (${recovery.testValidUntil}). Should be YYYY-MM-DD format`
```

Comment:
Note : EU required date format are like this;
<img width="634" alt="Screenshot 2022-05-15 at 10 14 40 PM" src="https://user-images.githubusercontent.com/10098074/168477284-84f231fa-3a19-4dc2-8717-1d1967e963ea.png">